### PR TITLE
Dchmyga/tacokit help content

### PR DESCRIPTION
**Pull request by chmyga**
**What is the current behavior?** (You can also link to an open issue here)
help content doesn't include TaCoKit components documentation

**What is the new behavior?**
help content includes TaCoKit components documentation automatically

**Please check if the PR fulfills these requirements**

- [ ] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitContribution.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+package org.talend.sdk.component.studio.documentation.toc;
+
+import org.eclipse.help.ITocContribution;
+import org.talend.sdk.component.studio.GAV;
+
+public class TaCoKitContribution implements ITocContribution {
+
+    private final String contributionName;
+    private String locale;
+    private TaCoKitIToc iToc;
```
Comment (gonchar-ivan) [inline main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitContribution.java]: something wrong with formatting. There should be empty line between fields

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitIToc.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    private final List<ITopic> topicsList = new ArrayList<>();
+    private String href;
+    private String label;
+
+    public TaCoKitIToc(final String href, final String label) {
+        this.href = href;
+        this.label = label;
+    }
+
+    public void addTopic(ITopic topic) {
```
Comment (gonchar-ivan) [inline main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitIToc.java]: final is missed

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitTocProvider.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+import org.talend.sdk.component.server.front.model.ComponentDetail;
+import org.talend.sdk.component.server.front.model.ComponentIndex;
+import org.talend.sdk.component.studio.Lookups;
+import org.talend.sdk.component.studio.i18n.Messages;
+import org.talend.sdk.component.studio.lang.Pair;
+import org.talend.sdk.component.studio.websocket.WebSocketClient;
+
+public class TaCoKitTocProvider extends AbstractTocProvider {
+
+    //Contributions by language.
```
Comment (gonchar-ivan) [inline main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitTocProvider.java]: why not javadoc?
`/** Contributions by language.*/`

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitTocProvider.java [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    private final Map<String, ITocContribution[]> languagePack = new HashMap<>();
+
+    @Override public ITocContribution[] getTocContributions(String language) {
```
Comment (gonchar-ivan) [inline main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitTocProvider.java]: final is missed

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitTocProvider.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        final Locale expLocale = getLocale(language);
+        
+        // let's build map of component families.
+        final Map<String, TaCoKitContribution> familyContributionsMap = new HashMap<>();
+        Stream<Pair<ComponentIndex, ComponentDetail>> details =
+                client.v1().component().details(expLocale.getLanguage());
+        details.forEach(pair -> {
+            final ComponentIndex index = pair.getFirst();
+            final String familyName = index.getFamilyDisplayName();
+            TaCoKitContribution familyContribution = familyContributionsMap.get(familyName);
```
Comment (gonchar-ivan) [inline main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitTocProvider.java]: consider
`familyContributionsMap.computeIfAbsent()`

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitTocProvider.java [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+import org.talend.sdk.component.studio.lang.Pair;
+import org.talend.sdk.component.studio.websocket.WebSocketClient;
+
+public class TaCoKitTocProvider extends AbstractTocProvider {
+
+    //Contributions by language.
+    private final Map<String, ITocContribution[]> languagePack = new HashMap<>();
+
+    @Override public ITocContribution[] getTocContributions(String language) {
+        ITocContribution[] contributions = languagePack.get(language);
```
Comment (gonchar-ivan) [inline main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitTocProvider.java]: consider computeIfAbsent() method here

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitTocProvider.java [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+            familyContribution.getToc().addTopic(topic);
+        });
+        
+        
+        contributions = familyContributionsMap.values().toArray(new ITocContribution[0]);
+        languagePack.put(language, contributions);
+        return contributions;
+    }
+
+    private Locale getLocale(final String locale) {
```
Comment (gonchar-ivan) [inline main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitTocProvider.java]: duplicated utility method.
Probably, extract it to utility class Locales.
getLocale() can be changed to static Locale newInstance(final String locale) or simply get(final String locale)

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitTopic.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+public class TaCoKitTopic implements ITopic {
+
+    private final List<ITopic> subTopics = new ArrayList<>();
+
+    private String label;
+
+    private String href;
+
+    public void addSubTopic(ITopic topic) {
```
Comment (gonchar-ivan) [inline main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitTopic.java]: final

**Comment by gonchar-ivan** [comment]
PR is good and can be merged. Comments above are related to code cleanliness only

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/context/TaCoKitContextProvider.java [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+public class TaCoKitContextProvider extends AbstractContextProvider {
+
+    @Override
+    public IContext getContext(String pluginId, String contextName) {
+        final Locale expLocale = getLocale(contextName);
+        final WebSocketClient client = Lookups.client();
+        //pluginId consists of two parts - plugin name and full component name and locale after the "."
+        //we will need to parse it to get the correct value of related topics.
+        final String fullComponentName = getFullComponentName(pluginId);
+        if (fullComponentName == null) {
```
[the remark this answers is not in the document]
Comment (chmyga) [reply main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/context/TaCoKitContextProvider.java]: In our case we need to return exactly null. Optional might not be supported by eclipse platform.

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/context/TaCoKitContextProvider.java [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+            }
+            result = input.substring(titleIndex + title.length(), descriptionEndIndex);
+            // if we don't have configuration title starting with "===", we might have table starting with "|==="
+            // thus we need to trim the "|" symbol.
+            if (result.endsWith("|")) {
+                result = result.substring(0, result.length() - 1);
+            }
+            // Now we need to check if we haven't added the next component to description. For that we need to look for
+            // "=="
+            if (result.indexOf("== ") > 0) {
```
[the remark this answers is not in the document]
Comment (chmyga) [reply main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/context/TaCoKitContextProvider.java]: thanks for this comment. I've forgotten to extract local variable.

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitContribution.java (hunk 1 above)
Comment (chmyga) [reply main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitContribution.java]: I think that I might be using a wrong formatter. Could you please tell which formatter do you use?

File: main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitTocProvider.java (hunk 6 above)
Comment (chmyga) [reply main/plugins/org.talend.sdk.component.studio-integration/src/main/java/org/talend/sdk/component/studio/documentation/toc/TaCoKitTocProvider.java]: we don't want to create any object there. so computeIfAbsent isn't a good option here.

**Pull request merged**